crates/sdk/src/masp/utils.rs

```diff
@@ -204,6 +709,7 @@ impl Iterator for FetchQueueReceiver {
                 if let Some(entry) = self.cache.pop_first() {
                     return Some(entry);
                 }
+                core::hint::spin_loop();
```

Comment:
Shouldn't this be handled by `flume`?

Reply:
the value here is being acquired through an indirection on a flume channel. we're accessing a cache that indirectly pulls from this channel. we could have used a condvar to check when a new value would be available on the cache, but the truth is we're replacing the entirety of this code in phase 3...